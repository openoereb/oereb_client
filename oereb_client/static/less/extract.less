/**
 * Extract styling
 */

@import "variables";

div#extract {
  width: @menu-with;
  max-width: 100%;
  position: fixed;
  top: 0;
  left: 0;
  height: 0;
  padding-top: @header-height;
  background-color: #fff;
  transition: height @animation-duration;
  overflow: hidden;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: stretch;
  div.extract-wrapper {
    z-index: 1;
    background-color: @menu-background-color;
    overflow-y: scroll;
    flex: 1;
    padding-bottom: @default-padding;
    margin-top: @default-padding;
  }
  span.extract-close {
    z-index: 2;
    display: block;
    float: right;
    color: @brand-primary;
    opacity: 1;
  }
  span.extract-close:hover {
    color: #000000;
    opacity: 1;
  }
  .panel-default > .panel-heading .badge {
    background-color: @brand-primary;
  }
  .collapse-extract {
    padding-top: @default-padding;
    .btn {
      border-bottom-width: 0;
    }
  }
}

div#extract.loading {
  height: @loading-indicator-height;
}

div#extract.active {
  height: 100%;
}

div#extract.information {
  box-shadow: @menu-shadow;
}