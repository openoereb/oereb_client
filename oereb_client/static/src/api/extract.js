export function queryExtractById(applicationUrl, egrid) {
    const url = new URL(applicationUrl + 'extract/reduced/json/geometry/' + egrid);
    url.searchParams.append('_dc', new Date().getTime());
    return fetch(url).then(response => response.json());
}

export function sanitizeTopicCode(code) {
    return code.replace('.', '').replace('-', '').replace('_', '');
}

export function groupRestrictionsByTopic(retstrictions) {
    const restrictionsByTopic = {};
    retstrictions.forEach((restriction) => {
        const code = sanitizeTopicCode(restriction.Theme.Code);
        if (!restrictionsByTopic.hasOwnProperty(code)) {
            restrictionsByTopic[code] = [];
        }
        restrictionsByTopic[code].push(restriction);
    });
    return restrictionsByTopic;
}
