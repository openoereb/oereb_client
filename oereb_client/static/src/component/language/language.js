import PropTypes from 'prop-types';
import React from "react";
import {withNamespaces} from 'react-i18next';
import {useDispatch, useSelector} from "react-redux";

import {setLanguage} from "../../reducer/language";

const OerebLanguage = function({t}) {
  const dispatch = useDispatch();
  const language = useSelector((state) => state.language);
  const currentLanguage = language.current;
  const availableLanguages = language.available;

  const selectLanguage = function(lang) {
    dispatch(setLanguage(lang));
  };

  const languages = availableLanguages.map((language, key) =>
    <li key={key}>
      <button className="dropdown-item" onClick={selectLanguage.bind(this, language)}>
        {language.toUpperCase()}
      </button>
    </li>
  );

  return (
    <div>
      <button className="btn btn-outline-secondary dropdown-toggle"
        type="button"
        data-bs-toggle="dropdown"
        aria-expanded="false"
        title={t('menu.language.title')}>
        {currentLanguage.toUpperCase()}
      </button>
      <ul className="dropdown-menu">
        {languages}
      </ul>
    </div>
  );
};

OerebLanguage.propTypes = {
  t: PropTypes.func.isRequired
};

export default withNamespaces()(OerebLanguage);
