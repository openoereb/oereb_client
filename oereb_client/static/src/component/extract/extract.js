import './extract.scss';

import React from 'react';
import { useSelector } from 'react-redux';
import OerebExtractLoading from './extract_loading';
import OerebExtractError from './extract_error';
import OerebExtractData from './extract_data';

function OerebExtract(props) {
    const extract = useSelector((state) => state.extract);

    function getContent(cfg) {
        if (cfg.loading) {
            return (
                <OerebExtractLoading />
            );
        }
        else if (cfg.error) {
            return (
                <OerebExtractError />
            );
        }
        else if (cfg.visible) {
            return (
                <OerebExtractData data={extract.data.GetExtractByIdResponse.extract} />
            );
        }
        else {
            return null;
        }
    }

    const content = getContent(extract);

    return (
        <div class="oereb-client-extract">
            {content}
        </div>
    );
}

export default OerebExtract;
