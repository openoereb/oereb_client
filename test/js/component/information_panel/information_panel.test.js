import {act, fireEvent, render, waitFor} from '@testing-library/react';
import React from "react";
import {Provider} from "react-redux";

import OerebInformationPanel
  from "../../../../oereb_client/static/src/component/information_panel/information_panel";
import {
  loadExtract,
  setInformationPanelTab,
  showExtract
} from "../../../../oereb_client/static/src/reducer/extract";
import {initLanguages} from "../../../../oereb_client/static/src/reducer/language";
import MainStore from "../../../../oereb_client/static/src/store/main";
import extract from "../../../../samples/extract.json";

let mockDate;

beforeAll(() => {
  mockDate = jest.spyOn(Date.prototype, 'toLocaleDateString').mockReturnValue('01.01.2022');
});

afterAll(() => {
  mockDate.mockRestore();
});

beforeEach(() => {
  act(() => {
    MainStore.dispatch(initLanguages({
      default: 'de',
      available: ['de']
    }));
  });
  act(() => {
    MainStore.dispatch(loadExtract({
      egrid: 'CH1234',
      zoom: false
    }));
  });
  act(() => {
    MainStore.dispatch(showExtract(extract));
  });
});

describe('information panel component', () => {

  let component;

  beforeEach(() => {
    component = render(
      <Provider store={MainStore}>
        <OerebInformationPanel />
      </Provider>
    );
  });

  it('should render general information', () => {
    act(() => {
      MainStore.dispatch(setInformationPanelTab(0));
    });
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should render disclaimer', () => {
    act(() => {
      MainStore.dispatch(setInformationPanelTab(1));
    });
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should render glossary', () => {
    act(() => {
      MainStore.dispatch(setInformationPanelTab(2));
    });
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should filter glossary results', async () => {
    act(() => {
      MainStore.dispatch(setInformationPanelTab(2));
    });
    fireEvent.change(component.container.querySelector('input'), {target: {value: 'kbs'}});
    await waitFor(() => {
      expect(component.asFragment()).toMatchSnapshot();
    });
  });

});