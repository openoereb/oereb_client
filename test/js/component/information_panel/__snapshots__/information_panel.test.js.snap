// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`information panel component should filter glossary results 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <OerebInformationPanel>
    <div
      className="oereb-client-information-panel container-fluid d-flex flex-column"
    >
      <div
        className="mb-4 oereb-client-information-panel-nav-mobile"
      >
        <div
          className="d-flex"
        >
          <ul
            className="nav nav-pills nav-justified flex-column"
          >
            <li
              className="nav-item"
              key="0"
            >
              <button
                className="nav-link"
                onClick={[Function]}
              >
                <span
                  className="text-body"
                >
                  extract.information_panel.general_information.title
                </span>
              </button>
            </li>
            <li
              className="nav-item"
              key="1"
            >
              <button
                className="nav-link"
                onClick={[Function]}
              >
                <span
                  className="text-body"
                >
                  extract.information_panel.exclusion_of_liability.title
                </span>
              </button>
            </li>
            <li
              className="nav-item"
              key="2"
            >
              <button
                className="nav-link active"
                onClick={[Function]}
              >
                <span
                  className="text-body"
                >
                  extract.information_panel.glossary.title
                </span>
              </button>
            </li>
          </ul>
          <div
            className="flex-grow-1 text-end ms-4"
          >
            <button
              className="btn btn-outline-secondary text-body"
              onClick={[Function]}
            >
              <i
                className="bi bi-x-lg"
              />
            </button>
          </div>
        </div>
        <div
          className="mt-2"
        >
          <input
            className="form-control"
            onChange={[Function]}
            placeholder="extract.information_panel.search.placeholder"
          />
        </div>
      </div>
      <div
        className="flex-grow-1 content-wrapper"
      >
        <OerebMultilingualCatalog
          catalog="Glossary"
          search="kbs"
        >
          <div
            className="container-fluid oereb-client-catalog"
          >
            <ul
              className="list-group list-group-flush"
            >
              <li
                className="list-group-item"
                key="0"
              >
                <p
                  className="fw-bold m-0"
                >
                  <span>
                    <span
                      key="0"
                    />
                    <mark
                      className="ps-0 pe-0"
                      key="1"
                    >
                      KbS
                    </mark>
                    <span
                      key="2"
                    />
                  </span>
                </p>
                <p
                  className="m-0"
                >
                  <span>
                    <span
                      key="0"
                    >
                      Kataster der belasteten Standorte
                    </span>
                  </span>
                </p>
              </li>
            </ul>
          </div>
        </OerebMultilingualCatalog>
      </div>
    </div>
  </OerebInformationPanel>
</Provider>
`;

exports[`information panel component should render disclaimer 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <OerebInformationPanel>
    <div
      className="oereb-client-information-panel container-fluid d-flex flex-column"
    >
      <div
        className="mb-4 oereb-client-information-panel-nav-mobile"
      >
        <div
          className="d-flex"
        >
          <ul
            className="nav nav-pills nav-justified flex-column"
          >
            <li
              className="nav-item"
              key="0"
            >
              <button
                className="nav-link"
                onClick={[Function]}
              >
                <span
                  className="text-body"
                >
                  extract.information_panel.general_information.title
                </span>
              </button>
            </li>
            <li
              className="nav-item"
              key="1"
            >
              <button
                className="nav-link active"
                onClick={[Function]}
              >
                <span
                  className="text-body"
                >
                  extract.information_panel.exclusion_of_liability.title
                </span>
              </button>
            </li>
            <li
              className="nav-item"
              key="2"
            >
              <button
                className="nav-link"
                onClick={[Function]}
              >
                <span
                  className="text-body"
                >
                  extract.information_panel.glossary.title
                </span>
              </button>
            </li>
          </ul>
          <div
            className="flex-grow-1 text-end ms-4"
          >
            <button
              className="btn btn-outline-secondary text-body"
              onClick={[Function]}
            >
              <i
                className="bi bi-x-lg"
              />
            </button>
          </div>
        </div>
        <div
          className="mt-2"
        >
          <input
            className="form-control"
            onChange={[Function]}
            placeholder="extract.information_panel.search.placeholder"
          />
        </div>
      </div>
      <div
        className="flex-grow-1 content-wrapper"
      >
        <OerebMultilingualCatalog
          catalog="Disclaimer"
          search={null}
        >
          <div
            className="container-fluid oereb-client-catalog"
          >
            <ul
              className="list-group list-group-flush"
            >
              <li
                className="list-group-item"
                key="0"
              >
                <p
                  className="fw-bold m-0"
                >
                  Eigentumsbeschränkungen im Grundbuch
                </p>
                <p
                  className="m-0"
                >
                  Zusätzlich zu den Angaben in diesem Auszug können Eigentumsbeschränkungen auch im Grundbuch angemerkt sein.
                </p>
              </li>
              <li
                className="list-group-item"
                key="1"
              >
                <p
                  className="fw-bold m-0"
                >
                  Haftungsausschluss Kataster der belasteten Standorte (KbS)
                </p>
                <p
                  className="m-0"
                >
                  Der Kataster der belasteten Standorte (KbS) wurde anhand der vom Bundesamt für Umwelt BAFU festgelegten Kriterien erstellt und wird fortwährend aufgrund neuer Erkenntnisse (z.B. Untersuchungen) aktualisiert. Die im KbS eingetragenen Flächen können vom tatsächlichen Ausmass der Belastung abweichen. Erscheint ein Grundstück nicht im KbS, besteht keine absolute Gewähr, dass das Areal frei von jeglichen Abfall- oder Schadstoffbelastungen ist. Bahnbetrieblich, militärisch und für die Luftfahrt genutzte Standorte liegen im Zuständigkeitsbereich des Bundes.
                </p>
              </li>
            </ul>
          </div>
        </OerebMultilingualCatalog>
      </div>
    </div>
  </OerebInformationPanel>
</Provider>
`;

exports[`information panel component should render general information 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <OerebInformationPanel>
    <div
      className="oereb-client-information-panel container-fluid d-flex flex-column"
    >
      <div
        className="mb-4 oereb-client-information-panel-nav-mobile"
      >
        <div
          className="d-flex"
        >
          <ul
            className="nav nav-pills nav-justified flex-column"
          >
            <li
              className="nav-item"
              key="0"
            >
              <button
                className="nav-link active"
                onClick={[Function]}
              >
                <span
                  className="text-body"
                >
                  extract.information_panel.general_information.title
                </span>
              </button>
            </li>
            <li
              className="nav-item"
              key="1"
            >
              <button
                className="nav-link"
                onClick={[Function]}
              >
                <span
                  className="text-body"
                >
                  extract.information_panel.exclusion_of_liability.title
                </span>
              </button>
            </li>
            <li
              className="nav-item"
              key="2"
            >
              <button
                className="nav-link"
                onClick={[Function]}
              >
                <span
                  className="text-body"
                >
                  extract.information_panel.glossary.title
                </span>
              </button>
            </li>
          </ul>
          <div
            className="flex-grow-1 text-end ms-4"
          >
            <button
              className="btn btn-outline-secondary text-body"
              onClick={[Function]}
            >
              <i
                className="bi bi-x-lg"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        className="flex-grow-1 content-wrapper"
      >
        <OerebGeneralInformation>
          <div
            className="container-fluid oereb-client-general-information"
          >
            <div
              className="card oereb-client-authority"
            >
              <div
                className="d-flex g-0 flex-column"
              >
                <div>
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      extract.information_panel.general_information.authority
                    </h5>
                    <p
                      className="card-text fw-bold m-0"
                    >
                      ÖREB-Katasteraufsichtsbehörde
                    </p>
                    <p
                      className="card-text m-0"
                    >
                      Seftigenstrasse 264
                    </p>
                    <p
                      className="card-text m-0"
                    >
                      3084 Wabern
                    </p>
                    <p
                      className="card-text m-0"
                    >
                      <a
                        href="https://www.cadastre.ch/en/oereb.html"
                        rel="noreferrer"
                        target="_blank"
                      >
                        https://www.cadastre.ch/en/oereb.html
                      </a>
                    </p>
                  </div>
                </div>
                <div>
                  <div
                    className="card-body"
                  >
                    <img
                      className="img-fluid oereb-client-authority-logo"
                      src="http://localhost:8080/samples/static/logo_canton.png"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              className="card mt-4 oereb-client-base-data"
            >
              <div
                className="d-flex g-0"
              >
                <div>
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      extract.information_panel.general_information.base_data
                    </h5>
                    <p
                      className="card-text m-0"
                    >
                      extract.information_panel.general_information.update_date_cs
                      : 
                      01.01.2022
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="card mt-4 oereb-client-general-information-card"
            >
              <div
                className="d-flex g-0"
              >
                <div>
                  <div
                    className="card-body pb-1"
                  >
                    <h5
                      className="card-title"
                    >
                      extract.information_panel.general_information.general_information
                    </h5>
                    <p
                      className="card-text m-0 mb-2"
                      key="0"
                    >
                      Der Inhalt des ÖREB-Katasters wird als bekannt vorausgesetzt. Der Kanton Basel-Landschaft ist für die Genauigkeit und Verlässlichkeit der gesetzgebenden Dokumente in elektronischer Form nicht haftbar. Der Auszug hat rein informativen Charakter und begründet insbesondere keine Rechte und Pflichten. Massgeblich sind diejenigen Dokumente, welche rechtskräftig verabschiedet oder veröffentlicht worden sind. Weitere Informationen zum ÖREB-Kataster finden Sie unter www.cadastre.ch.
                    </p>
                    <p
                      className="card-text m-0 mb-2"
                      key="1"
                    >
                      Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="card mt-4 oereb-client-involved-parties"
            >
              <div
                className="d-flex g-0"
              >
                <div>
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      extract.information_panel.general_information.further_participants
                    </h5>
                    <div
                      className="card-text d-flex align-items-stretch flex-column"
                    >
                      <div
                        className="text-center"
                      >
                        <img
                          className="img-fluid oereb-client-federal-logo"
                          src="http://localhost:8080/samples/static/logo_confederation.png"
                        />
                      </div>
                      <div
                        className="text-center"
                      >
                        <img
                          className="img-fluid oereb-client-municipality-logo"
                          src="http://localhost:8080/samples/static/diegten.png"
                        />
                        <p>
                          Oberwil (BL)
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </OerebGeneralInformation>
      </div>
    </div>
  </OerebInformationPanel>
</Provider>
`;

exports[`information panel component should render glossary 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <OerebInformationPanel>
    <div
      className="oereb-client-information-panel container-fluid d-flex flex-column"
    >
      <div
        className="mb-4 oereb-client-information-panel-nav-mobile"
      >
        <div
          className="d-flex"
        >
          <ul
            className="nav nav-pills nav-justified flex-column"
          >
            <li
              className="nav-item"
              key="0"
            >
              <button
                className="nav-link"
                onClick={[Function]}
              >
                <span
                  className="text-body"
                >
                  extract.information_panel.general_information.title
                </span>
              </button>
            </li>
            <li
              className="nav-item"
              key="1"
            >
              <button
                className="nav-link"
                onClick={[Function]}
              >
                <span
                  className="text-body"
                >
                  extract.information_panel.exclusion_of_liability.title
                </span>
              </button>
            </li>
            <li
              className="nav-item"
              key="2"
            >
              <button
                className="nav-link active"
                onClick={[Function]}
              >
                <span
                  className="text-body"
                >
                  extract.information_panel.glossary.title
                </span>
              </button>
            </li>
          </ul>
          <div
            className="flex-grow-1 text-end ms-4"
          >
            <button
              className="btn btn-outline-secondary text-body"
              onClick={[Function]}
            >
              <i
                className="bi bi-x-lg"
              />
            </button>
          </div>
        </div>
        <div
          className="mt-2"
        >
          <input
            className="form-control"
            onChange={[Function]}
            placeholder="extract.information_panel.search.placeholder"
          />
        </div>
      </div>
      <div
        className="flex-grow-1 content-wrapper"
      >
        <OerebMultilingualCatalog
          catalog="Glossary"
          search={null}
        >
          <div
            className="container-fluid oereb-client-catalog"
          >
            <ul
              className="list-group list-group-flush"
            >
              <li
                className="list-group-item"
                key="0"
              >
                <p
                  className="fw-bold m-0"
                >
                  Baulinien Eisenbahnanlagen
                </p>
                <p
                  className="m-0"
                >
                  Zwischen den Baulinien dürfen keine baulichen Veränderungen vorgenommen oder sonstige Massnahmen getroffen werden, die dem Zweck bestehender oder künftiger Eisenbahnanlagen widersprechen.
                </p>
              </li>
              <li
                className="list-group-item"
                key="1"
              >
                <p
                  className="fw-bold m-0"
                >
                  Baulinien Flughafenanlagen
                </p>
                <p
                  className="m-0"
                >
                  Zwischen den Baulinien dürfen keine baulichen Veränderungen vorgenommen oder sonstige Massnahmen getroffen werden, die dem Zweck bestehender oder künftiger Flughafenanlagen widersprechen.
                </p>
              </li>
              <li
                className="list-group-item"
                key="2"
              >
                <p
                  className="fw-bold m-0"
                >
                  Baulinien Nationalstrassen
                </p>
                <p
                  className="m-0"
                >
                  Wenn der projektierte Strassenverlauf definitiv bekannt ist, werden beiderseits der Strasse Baulinien festgelegt. Diese Baulinien ermöglichen es, die Anforderungen der Verkehrssicherheit und der Wohnhygiene sowie die Erfordernisse eines eventuellen künftigen Ausbaus der Strasse zu berücksichtigen. Zwischen den Baulinien dürfen ohne Bewilligung weder Neubauten erstellt noch Umbauten bestehender Gebäude vorgenommen werden, auch wenn diese nur teilweise in die Baulinien hineinragen.
                </p>
              </li>
              <li
                className="list-group-item"
                key="3"
              >
                <p
                  className="fw-bold m-0"
                >
                  Baulinien Starkstromanlagen
                </p>
                <p
                  className="m-0"
                >
                  Zwischen den Baulinien dürfen keine baulichen Veränderungen vorgenommen oder sonstige Massnahmen getroffen werden, die dem Zweck bestehender oder künftiger Starkstromanlagen widersprechen.
                </p>
              </li>
              <li
                className="list-group-item"
                key="4"
              >
                <p
                  className="fw-bold m-0"
                >
                  BFS-Nr.
                </p>
                <p
                  className="m-0"
                >
                  Gemeindenummer aus amtlichem Gemeindeverzeichnis; Eine vom schweizerischen Bundesamt für Statistik mit dem Amtlichen Gemeindeverzeichnis im Jahre 1986 erstmals vergebene Zahl, die der eineindeutigen Bezeichnung von territorialen Einheiten im Einzugsbereich der Schweiz dient.
                </p>
              </li>
              <li
                className="list-group-item"
                key="5"
              >
                <p
                  className="fw-bold m-0"
                >
                  E-GRID
                </p>
                <p
                  className="m-0"
                >
                  Eidgenössischer Grundstücksidentifikator; Aus einem Präfix und einer Nummer bestehende Bezeichnung, die es erlaubt, jedes in das Grundbuch aufgenommene Grundstück landesweit eindeutig zu identifizieren, und die zum Datenaustausch zwischen Informatiksystemen dient.
                </p>
              </li>
              <li
                className="list-group-item"
                key="6"
              >
                <p
                  className="fw-bold m-0"
                >
                  Eigentumsbeschränkung
                </p>
                <p
                  className="m-0"
                >
                  Der Zweck aller Eigentumsbeschränkungen ist die Wahrung von Interessen anderer Personen, denen jene des Eigentümers sich in bestimmter Beziehung unterordnen müssen. Dies sind entweder die Eigentümer angrenzender Grundstücke, die Nachbarn, oder auch weitere Kreise von Privatpersonen oder endlich die Allgemeinheit selber, der Staat. Die Beschränkungen zugunsten der Nachbarn oder weiterer Privatpersonen sind regelmässig privatrechtlicher, jene zugunsten der Allgemeinheit öffentlich-rechtlicher Natur.
                </p>
              </li>
              <li
                className="list-group-item"
                key="7"
              >
                <p
                  className="fw-bold m-0"
                >
                  Gesetzliche Grundlage
                </p>
                <p
                  className="m-0"
                >
                  Gesetz, Verordnung etc., das generell-abstrakt ist (generell für die Person, die nicht bekannt ist, abstrakt, weil der Perimeter ohne Karte definiert ist) und auf Bundesebene, auf kantonaler oder kommunaler Ebene erlassen worden ist und die bloss eine allgemeine Rechtsgrundlage der Eigentumsbeschränkung darstellen. Die gesetzliche Grundlage ist nicht Teil des ÖREB-Katasters. Der ÖREB-Kataster enthält aber Hinweise auf die entsprechende gesetzliche Grundlage.
                </p>
              </li>
              <li
                className="list-group-item"
                key="8"
              >
                <p
                  className="fw-bold m-0"
                >
                  Gewässerraum
                </p>
                <p
                  className="m-0"
                >
                  Fliessgewässer können nur wieder naturnäher werden, wenn ausreichend Raum in den Schutz der Gewässer miteinbezogen wird. Der Gewässerraum soll weitgehend frei von neuen Anlagen bleiben; bestehende Anlagen haben jedoch Bestandesgarantie.
                </p>
              </li>
              <li
                className="list-group-item"
                key="9"
              >
                <p
                  className="fw-bold m-0"
                >
                  Grundwasserschutzareale
                </p>
                <p
                  className="m-0"
                >
                  Von Kanton festgelegte Areale, die für die künftige Nutzung und Anreicherung von Grundwasservorkommen von Bedeutung sind und in dem keine Bauten und Anlagen erstellt oder Arbeiten ausgeführt werden, die künftige Nutzungs- und Anreicherungsanlagen beeinträchtigen könnten.
                </p>
              </li>
              <li
                className="list-group-item"
                key="10"
              >
                <p
                  className="fw-bold m-0"
                >
                  Grundwasserschutzzonen
                </p>
                <p
                  className="m-0"
                >
                  Grundwasserschutzzonen, in denen gewährleistet werden soll, dass bei unmittelbar drohenden Gefahren wie Unfällen mit wassergefährdenden Stoffen ausreichend Zeit und Raum für die erforderlichen Massnahmen zur Verfügung stehen.
                </p>
              </li>
              <li
                className="list-group-item"
                key="11"
              >
                <p
                  className="fw-bold m-0"
                >
                  Kataster der belasteten Standorte
                </p>
                <p
                  className="m-0"
                >
                  Der Kataster enthält Deponien und andere durch Abfälle belastete Standorte, die saniert werden, wenn sie zu schädlichen oder lästigen Einwirkungen führen oder die konkrete Gefahr besteht, dass solche Einwirkungen entstehen. Die Behörde ermittelt die belasteten Standorte, indem sie vorhandene Angaben wie Karten, Verzeichnisse und Meldungen auswertet. Sie trägt diejenigen Standorte, deren Belastung erwiesen oder sehr wahrscheinlich ist, in den Kataster ein.
                </p>
              </li>
              <li
                className="list-group-item"
                key="12"
              >
                <p
                  className="fw-bold m-0"
                >
                  Katasterverantwortliche Stelle
                </p>
                <p
                  className="m-0"
                >
                  Die katasterverantwortliche Stelle im Kanton erhält die in den ÖREB-Kataster aufzunehmenden Daten von den zuständigen Fachstellen. Sie verwaltet diese Daten und stellt sie via kantonales ÖREB-Geoportal der Öffentlichkeit zur Verfügung.
                </p>
              </li>
              <li
                className="list-group-item"
                key="13"
              >
                <p
                  className="fw-bold m-0"
                >
                  KbS
                </p>
                <p
                  className="m-0"
                >
                  Kataster der belasteten Standorte
                </p>
              </li>
              <li
                className="list-group-item"
                key="14"
              >
                <p
                  className="fw-bold m-0"
                >
                  Lärmempfindlichkeitsstufen
                </p>
                <p
                  className="m-0"
                >
                  Empfindlichkeitsstufen werden festgelegt, um jeweils bestimmte Zonen zu definieren: diejenigen, die eines erhöhten Lärmschutzes bedürfen, diejenigen, in denen keine störenden Betriebe zugelassen sind, und diejenigen, in denen mässig und stark störende Betriebe zugelassen sind.
                </p>
              </li>
              <li
                className="list-group-item"
                key="15"
              >
                <p
                  className="fw-bold m-0"
                >
                  Nutzungsplanung
                </p>
                <p
                  className="m-0"
                >
                  Festlegung der Verwendung einzelner Bodenflächen für bestimmte Zwecke (z. B. Landwirtschaft, Siedlung, Wald).
                </p>
              </li>
              <li
                className="list-group-item"
                key="16"
              >
                <p
                  className="fw-bold m-0"
                >
                  ÖREB
                </p>
                <p
                  className="m-0"
                >
                  Öffentlich-rechtliche Eigentumsbeschränkung
                </p>
              </li>
              <li
                className="list-group-item"
                key="17"
              >
                <p
                  className="fw-bold m-0"
                >
                  ÖREB-Kataster
                </p>
                <p
                  className="m-0"
                >
                  Kataster der öffentlich-rechtlichen Eigentumsbeschränkungen
                </p>
              </li>
              <li
                className="list-group-item"
                key="18"
              >
                <p
                  className="fw-bold m-0"
                >
                  Planungszonen
                </p>
                <p
                  className="m-0"
                >
                  Zone, die ein genau bezeichnetes Gebiet umfasst, für das ein Nutzungsplan angepasst werden muss oder noch keiner vorliegt, und innerhalb derer nichts unternommen werden darf, was die Nutzungsplanung erschwert.
                </p>
              </li>
              <li
                className="list-group-item"
                key="19"
              >
                <p
                  className="fw-bold m-0"
                >
                  Projektierungszonen Eisenbahnanlagen
                </p>
                <p
                  className="m-0"
                >
                  Um die freie Verfügbarkeit der für künftige Bahnbauten und -anlagen erforderlichen Grundstücke zu gewährleisten, können für genau bezeichnete Gebiete Projektierungszonen festgelegt werden. In diesen Zonen dürfen keine baulichen Veränderungen vorgenommen werden, die dem Zweck der Eisenbahnen widersprechen.
                </p>
              </li>
              <li
                className="list-group-item"
                key="20"
              >
                <p
                  className="fw-bold m-0"
                >
                  Projektierungszonen Flughafenanlagen
                </p>
                <p
                  className="m-0"
                >
                  Um die freie Verfügbarkeit der für Flughafenanlagen erforderlichen Grundstücke zu gewährleisten, können für genau bezeichnete Gebiete Projektierungszonen festgelegt werden. In diesen Zonen dürfen keine baulichen Veränderungen vorgenommen werden, die dem Zweck der Flughafenanlagen widersprechen.
                </p>
              </li>
              <li
                className="list-group-item"
                key="21"
              >
                <p
                  className="fw-bold m-0"
                >
                  Projektierungszonen Leitungen mit einer Nennspannung von 220 kV oder höher
                </p>
                <p
                  className="m-0"
                >
                  Um die freie Verfügbarkeit der für Flughafenanlagen erforderlichen Grundstücke zu gewährleisten, können für genau bezeichnete Gebiete Projektierungszonen festgelegt werden. In diesen Zonen dürfen keine baulichen Veränderungen vorgenommen werden, die dem Zweck der Flughafenanlagen widersprechen.
                </p>
              </li>
              <li
                className="list-group-item"
                key="22"
              >
                <p
                  className="fw-bold m-0"
                >
                  Projektierungszonen Nationalstrassen
                </p>
                <p
                  className="m-0"
                >
                  Um die freie Verfügbarkeit des für den Bau der Nationalstrassen erforderlichen Landes zu gewährleisten, können für genau bezeichnete Gebiete Projektierungszonen festgelegt werden. In diesen Zonen dürfen keine baulichen Veränderungen vorgenommen werden, die dem Zweck der Nationalstrassen widersprechen.
                </p>
              </li>
              <li
                className="list-group-item"
                key="23"
              >
                <p
                  className="fw-bold m-0"
                >
                  Rechtsvorschrift
                </p>
                <p
                  className="m-0"
                >
                  Generell-konkrete Rechtsnorm, die zusammen mit den ihr zugeschriebenen Geobasisdaten die Eigentumsbeschränkung unmittelbar umschreibt und innerhalb desselben Verfahrens verabschiedet worden ist.
                </p>
              </li>
              <li
                className="list-group-item"
                key="24"
              >
                <p
                  className="fw-bold m-0"
                >
                  Sicherheitszonenplan
                </p>
                <p
                  className="m-0"
                >
                  Zonenplan, in dem die Sicherheitszonen dargestellt sind und aus dem die Eigentumsbeschränkungen nach Fläche und Höhe ersichtlich sind.
                </p>
              </li>
              <li
                className="list-group-item"
                key="25"
              >
                <p
                  className="fw-bold m-0"
                >
                  Statische Waldgrenzen
                </p>
                <p
                  className="m-0"
                >
                  Statische Waldgrenzen müssen auf der Grundlage rechtskräftiger Waldfeststellungen festgelegt werden. Neue Bestockungen ausserhalb dieser Waldgrenzen gelten nicht als Wald.
                </p>
              </li>
              <li
                className="list-group-item"
                key="26"
              >
                <p
                  className="fw-bold m-0"
                >
                  Vorwirkung
                </p>
                <p
                  className="m-0"
                >
                  Mit der öffentlichen Auflage entsteht eine Vorwirkung: Ab diesem Zeitpunkt dürfen Bauvorhaben in der Regel nur bewilligt werden, wenn sie der öffentlich aufgelegten Planung nicht widersprechen.
                </p>
              </li>
              <li
                className="list-group-item"
                key="27"
              >
                <p
                  className="fw-bold m-0"
                >
                  Waldabstandslinien
                </p>
                <p
                  className="m-0"
                >
                  Bauten und Anlagen in Waldesnähe sind nur zulässig, wenn sie die Erhaltung, Pflege und Nutzung des Waldes nicht beeinträchtigen.
                </p>
              </li>
              <li
                className="list-group-item"
                key="28"
              >
                <p
                  className="fw-bold m-0"
                >
                  Waldreservate
                </p>
                <p
                  className="m-0"
                >
                  Geschützte Waldfläche, die der Erhaltung der Artenvielfalt von Fauna und Flora dient.
                </p>
              </li>
              <li
                className="list-group-item"
                key="29"
              >
                <p
                  className="fw-bold m-0"
                >
                  Zuständige Stelle
                </p>
                <p
                  className="m-0"
                >
                  Durch die Gesetzgebung bezeichnete Stelle des Bundes, des Kantons oder der Gemeinde, die für das Erheben, Nachführen und Verwalten der Geobasisdaten zuständig ist.
                </p>
              </li>
            </ul>
          </div>
        </OerebMultilingualCatalog>
      </div>
    </div>
  </OerebInformationPanel>
</Provider>
`;
